src/Axe/Managers/CompilerManager.cs

```diff
@@ -265,10 +276,105 @@ public CompilerManager(AxeDbContext context, INodeServices nodeServices) : base(
                 return new Tuple<CodeBlockResult, string[]>(CodeBlockResult.Error, new string[] { exception.Message });
             }
         }
-        
+
+        private async Task<Tuple<CodeBlockResult, string[]>> InterpretPython(CodeBlockSolveVm model)
+        {
+            try
+            {
+                CodeBlock codeBlock = context.CodeBlock
+                    .Include(cb => cb.TestCases)
+                    .First(cb => cb.Id == model.Id);
+                List<string> failedTestCases = new List<string>();
+
+                string pyFilePath = Path.GetTempPath() + Path.GetRandomFileName().Replace(".", "") + ".py";
+
+                //ProcessStartInfo startInfo = new ProcessStartInfo("C:\\Users\\andrei.shilkin\\AppData\\Local\\Programs\\Python\\Python36-32\\python.exe");
+                ProcessStartInfo startInfo = new ProcessStartInfo(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) + "\\AppData\\Local\\Programs\\Python\\Python36-32\\python.exe");
+                startInfo.Arguments = pyFilePath;
+                startInfo.UseShellExecute = false;
+                startInfo.CreateNoWindow = true;
+                startInfo.RedirectStandardOutput = true;
+                startInfo.RedirectStandardError = true;
+
+                for (int i = 0; i < codeBlock.TestCases.Count; i++)
+                {
+                    // Generate the code for execution
+                    string code = model.SourceCode;
+                    if (codeBlock.TestCases[i].Input != null)
+                    {
+                        string[] inputArguments = codeBlock.TestCases[i].Input.Split(' ');
+                        code += "\nprint(main(";
+                        for (int j = 0; j < inputArguments.Length; j++)
+                        {
+                            code += inputArguments[j];
+                            if (j < inputArguments.Length - 1)
+                            {
+                                code += ", ";
+                            }
+                        }
```

Comment:
you REALLY do not want to concat many strings in a loop. memory is allocated for all intermediate results ("1,", then "1,2,", "1,2,3,", etc). with a large number of items you might get a performance hit.
if you are going to build a string from many words which maybe different - then use StringBuilder: `var sb = new StringBuilder(); sb.AppendLine("smth"); sb.AppendLine("another");`
for this case - String.Join() can replace entire loop

`code += string.Join(",", inputArguments);`

last time I checked - string.Join uses StringBuilder inside